feat: Add PendingRequestsView and category icons

BarreiroReSird opened the pull request:
Introduces the PendingRequestsView composable for displaying pending employee requests, along with new vector drawable icons for request categories (basket, cutlery, mop, perfume). Also removes an unused deployment target selection from project configuration.

BarreiroReSird committed:
feat: Add QueuedRequestsView and category icons

Introduces the QueuedRequestsView composable for displaying pending employee requests, along with new vector drawable icons for request categories (basket, cutlery, mop, perfume). Also removes an unused deployment target selection from project configuration.

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
Se for para os "Pedidos pendentes" mete o nome como "PendingRequestsView" e não "Queued"

Basiiii commented on app/.idea/deploymentTargetSelector.xml:
Acho que não é preciso mudar este ficheiro

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+import androidx.compose.ui.unit.sp
+import com.lojasocial.app.R
+
+// Cores ajustadas para maior fidelidade ao design
```
Tira comentários Gemini e mete as cores no ficheiro do Tema (theme) da app

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+val ColorDivider = Color(0xFFF2F4F7)
+val ColorArrow = Color(0xFFD0D5DD)
+
+data class QueuedRequest(
```
Modelos devem estar num ficheiro à parte

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun QueuedRequestsView(
```
Quando mudares o nome do ficheiro (para tirar o "Queue"), muda o nome da função

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+fun QueuedRequestsView(
+    onBack: () -> Unit = {}
+) {
+    val requests = listOf(
```
Mete os dados a vir da base de dados (podes meter esta simulação num preview)

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+            .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically // Garante que avatar, texto e seta alinhem pelo centro
+    ) {
+        // Avatar Placeholder (Como solicitado, mantido, mas com cor suave)
```
Tira comentários Gemini

Basiiii commented on app/app/src/main/java/com/lojasocial/app/ui/employees/QueuedRequestsView.kt:
```diff
@@ -0,0 +1,238 @@
+}
+
+@Composable
+fun StatusBadge() {
```
Mete estas funções em ficheiros separados (componentes)

BarreiroReSird committed:
feat: Add pending requests feature for employees

Introduced PendingRequestsRepository, domain model, UI state, ViewModel, and new composable views to display pending requests for employees. Replaced QueuedRequestsView with a new implementation using PendingRequest and modularized RequestItem and StatusBadge components. Updated DI module for repository injection and adjusted project configuration files.

BarreiroReSird committed:
refactor: Refactor pending requests feature package structure

Moved PendingRequests-related classes from 'employees' to 'requests' and 'requests/components' packages for better organization. Updated repository import paths and renamed AppModule to PendingRequestsModule to reflect the new structure.

BarreiroReSird commented:
@Basiiii
